### PR TITLE
DRIVERS-1591 clarify comparison behaviour for version strings

### PR DESCRIPTION
**Pull request by alcaeus**
DRIVERS-1591

### CODE REVIEW

File: source/unified-test-format/unified-test-format.rst [hunk 1]
```diff
@@ -1044,6 +1044,12 @@ non-negative integer:
 - ``<major>.<minor>`` (``<patch>`` is assumed to be zero)
 - ``<major>`` (``<minor>`` and ``<patch>`` are assumed to be zero)
 
+Any component other than ``major``, ``minor``, and ``patch`` MUST be discarded
```
Comment (jyemin) [inline source/unified-test-format/unified-test-format.rst:1047]: Consider whether to be more proscriptive of how the actual server version is obtained.  If the spec says explicitly that it's obtained by taking the first three elements of the `versionArray` field in the `buildInfo` reply, we can avoid any mention of discarding other components.

**Review by DmitryLukyanov — APPROVED** [review]
LGTM

File: source/unified-test-format/unified-test-format.rst (hunk 1 above)
Comment (alcaeus) [reply source/unified-test-format/unified-test-format.rst:1047]: I've expanded the wording to explain how to obtain this information from the `versionArray` field.

**Review by jyemin — APPROVED** [review]
LGTM

**Pull request merged**